### PR TITLE
With services knob

### LINKED ISSUE #2374: default service in input provider doesn't work
**Describe the bug**
Setting a default service in the input provider doesn't have any effect and the service will always default to `news`.

**To Reproduce**
Steps to reproduce the behaviour:

1. Go to `psammead/packages/components/psammead-paragraph/src/index.stories.jsx` and add `options: { defaultService: 'arabic' }` to the `inputProvider` function call.
2. Go to http://localhost:8180/?path=/story/components-paragraph--default
3. Observe that the default service hasn't taken effect and still defaults to `news`

**Expected behaviour**
The default service should work correctly. I don't think it will be possible to work in it's current implementation. We should consider refactoring input provider into a decorator as demonstrated with `withServicesKnob` in https://github.com/bbc/psammead/pull/2366/files#diff-696b02495ed3b190dee543b663004b46
**Testing notes**

- [ ] This bug fix is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by jroebu14**
Resolves https://github.com/bbc/psammead/issues/2374 and https://github.com/bbc/psammead/issues/2371

**Overall change:** Adds a function called `withServicesKnob` that returns a storybook decorator function that does 2 things:

1. Provides the decorated stories with the following properties that can be passed into components:

- `text`: A string of text in the language of the chosen service.
- `dir`: The reading directionality of the chosen service e.g. `ltr` or `rtl`
- `script`: The chosen service's script typography settings e.g. the font-size and line-heights.
- `service`: The name of the chosen service e.g. `arabic`

2. Toggles the layout directionality of the chosen service.

The `withServicesKnob` function accepts an options argument with 2 properties:

- `defaultService`(String): The default selected service of the services dropdown e.g. `arabic`. The default is `news`.
- `services`(Array): A list of services that the dropdown will display. The default is all services.

`withServicesKnob` will eventually supersede all use of `inputProvider` and `withDir` as it combines and simplifies their functionality and fixes the bugs https://github.com/bbc/psammead/issues/2374 and https://github.com/bbc/psammead/issues/2371 i.e. the default service bug and the broken text slots API.

**Code changes:**

- Adds new feature `withServicesKnob` to `psammead-storybook-helpers`
- Updates readme with documentation of new feature
- Bumps the package.json minor version and update changelog

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Automated jest tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

### CODE REVIEW

**Review by oluoluoxenfree — APPROVED** [review]

File: packages/utilities/psammead-storybook-helpers/README.md [hunk 1]
```diff
@@ -21,7 +21,53 @@ This package provides a collection of common values that are used in storybook b
...
+      </Paragraph>
+    ),
+```
+
+To set a default service:
+
+```js
+storiesOf('Components|Paragraph', module)
+  .addDecorator(withKnobs)
+  .addDecorator(withServicesKnob('arabic', ['news', 'arabic', 'amharic'])); // default selected service is `arabic` and the available services in the dropdown are `news`, `arabic`, `amharic`
```
Comment (ryanmccombe) [inline packages/utilities/psammead-storybook-helpers/README.md]: this argument list might work better as an object - won't be clear from reading code what the args mean, can't pass a service list without setting default service, and will be more problematic if we add more options in the future

**Review by ryanmccombe — APPROVED** [review]

File: packages/utilities/psammead-storybook-helpers/README.md (hunk 1 above)
Comment (jroebu14) [reply packages/utilities/psammead-storybook-helpers/README.md]: I was literally this minute thinking that! Cool I'll update this

**Review by EinsteinNjoroge — APPROVED** [review]
LGTM though you might wanna consider adding tests for `withServicesKnob.jsx`

**Review by FK78 — APPROVED** [review]
LGTM.

**Comment by jroebu14** [comment]
@EinsteinCarrey @ryanmccombe unit tests added https://github.com/bbc/psammead/pull/2402/commits/08343f16ac0fda0f1d14e2e4b5cb19c3e1ed8ca8

**Review by dr3 — CHANGES_REQUESTED** [review]
Can you add examples to `psammead-storybook-helpers/src/index.stories.jsx` like inputProvider has please?

**Comment by jroebu14** [comment]
> Can you add examples to `psammead-storybook-helpers/src/index.stories.jsx` like inputProvider has please?

@dr3 Done!

**Review by dr3 — CHANGES_REQUESTED** [review]
If I change from a `rtl` service to a `ltr` one it updates, but after that, changing it again doesnt work, it stays on the 2nd service forever

**Comment by jroebu14** [comment]
> If I change from a rtl service to a ltr one it updates, but after that, changing it again doesnt work, it stays on the 2nd service forever

@dr3 fixed this issue in https://github.com/bbc/psammead/pull/2402/commits/807228e98e18931ddb195081ab48c009643e6cb1

**Review by dr3 — APPROVED** [review]

**Pull request merged**